test/test-lex.cpp

```diff
@@ -221,27 +345,25 @@ TEST(test_lex, lex_invalid_big_int_number) {
                 VariantWith<error_big_int_literal_contains_decimal_point>(_)));
       });
 
-  // Complain about both the decimal point and the leading 0 digit.
+  /* // Complain about both the decimal point and the leading 0 digit. */
   check_tokens_with_errors(
-      u8"01.2n", {token_type::number},
+      u8"01.2n", {token_type::number,token_type::number},
       [](padded_string_view, const auto& errors) {
         EXPECT_THAT(
             errors,
             UnorderedElementsAre(
```

Comment:
Minor: `ElementsAre` produces better error messages than `UnorderedElementsAre`. Prefer the former where possible.